
3164700222
375252

# bulk-seq

# Immune infiltration and clinical significance analyses of the coagulation-related genes in hepatocellular carcinoma.

# Database (Methy)

<https://cloud.tencent.com/developer/article/1966159>

# GWAS database

<https://www.ebi.ac.uk/gwas/>

# TWAS

TWAS Atlas: a curated knowledgebase of transcriptome-wide association studies
<https://pubmed.ncbi.nlm.nih.gov/36243959/>

# Causal genes

Identifying causal genes for migraine by integrating the proteome and transcriptome (PMID:37592229)
<https://pmc.ncbi.nlm.nih.gov/articles/PMC9022281/>

- PWAS, ?
- TWAS， (GWAS + FUSION (PMID:26854917) <http://gusevlab.org/projects/fusion/>) (<https://github.com/gusevlab/fusion_twas>)
- TWAS， (GWAS + S-PrediXcan (PMID:29739930) <https://github.com/hakyimlab/MetaXcan> + FOCUS (PMID:31844974) <https://github.com/ChongWu-Biostat/FOGS> ) 

# Thought

- pan-cancer TCGA
- Causal genes (<https://pubmed.ncbi.nlm.nih.gov/37592229/>, PMID:37592229, IF:7.28)
- Mitochondrial dysfunction, & cancer (<https://pubmed.ncbi.nlm.nih.gov/36634566/>, PMID:36634566, IF:8.14)

# scRNA-seq prediction metabolic flux


[@AGraphNeuralAlgham2021]
scFEA: A graph neural network model to estimate cell-wise metabolic using single cell RNA-seq data
- <https://github.com/changwn/scFEA>

# protein-protein docking

RosettaDock4.0

- <https://zhuanlan.zhihu.com/p/490274309>
- <https://pubmed.ncbi.nlm.nih.gov/17825317/>
- <https://pubmed.ncbi.nlm.nih.gov/18640688/>
- <https://www.rosettacommons.org/docs/latest/getting_started/Getting-Started>
- <https://github.com/RosettaCommons/rosetta>
- <https://www.pyrosetta.org/downloads>
- <https://cluspro.bu.edu/publications.php>


# OpenAI

thlisintralin1974@hotmail.com
CKZ0ligD54034

<https://sms-activate.org/cn>

# 基因序列

<https://zhuanlan.zhihu.com/p/557609219>

# Single-cell protein activity

<https://github.com/califano-lab/PISCES/blob/master/vignettes/pisces_pbmc.Rmd>

# Single-cell RNA-seq with Bulk seq

- Bayesian cell Proportion Reconstruction <https://github.com/Danko-Lab/BayesPrism>
- <https://github.com/sunduanchen/Scissor>

# Single Cell ATAC

Single-cell multi-ome regression models identify functional and
disease-associated enhancers and enable chromatin potential analysis
<https://www.nature.com/articles/s41588-024-01689-8?fromPaywallRec=false>

# Single Cell annotation

Assessing GPT-4 for cell type annotation in single-cell RNA-seq analysis
<https://www.nature.com/articles/s41592-024-02235-4?fromPaywallRec=false>

# GWAS

## Concept

等位基因：对于二倍体，一对染色体中的1号和2号的相同基因座对应的基因。可以为不同形式。

基因座（英語：locus，gene locus；複數：loci）又稱基因位點、位點，是一生物學與遺傳算法術語，指某個基因或具有調控作用的遺傳標記在染色體上所處的特定位置。一個基因座可用染色體編號、染色體長短臂代號和其在染色體上所處的「區」、「帶」、「子帶」編號位置表示。

基因座上的DNA序列可能有許多不同的變化，各種變化形式稱為等位基因（allele）。基因座在基因組中的排列位置稱為基因圖譜（genetic map），基因作圖（gene mapping）則是測定基因座與特定性狀關係的過程。

二倍體與多倍體細胞的某些染色體上，在同一基因座上有相同的等位基因，這類細胞稱為純合子/同型合子（homozygous）。若是相同基因座上含有不同的等位基因，則稱作雜合子/異型合子（heterozygous）。

连锁不平衡（Linkage Disequilibrium，LD）指的是在某一群体中，分属两个或两个以上基因座位的等位基因同时出现的几率高于或低于随机出现的几率的现象。只要两个基因不是完全独立地遗传，就会表现出某种程度的连锁不平衡。这种不平衡可以是同一条染色体上的不同区域，也可以是不同染色体上的。例如，如果两个基因座位的等位基因以一定的频率在人群中出现，而这些基因并非完全随机地组成单体型，就会导致某些单体型

在同源染色体相同位置上存在三种以上不同形式的等位基因,称为复等位基因.
这是指1个群体的情况,对1个个体来说,在同源染色体相同位置上最多含其中的2种,比如人的ABO血型:控制血型的基因有3个,但每一个人的血型基因最多有2种

重叠基因（Overlapping genes），亦作“多重编码”（dual coding），是指两个或两个以上的基因共享一段DNA序列的现象

多基因评分（Polygenic Scores，简称PGS；也被称为Polygenic Risk Scores或PRS）是与特定性状（包括行为、特征或疾病）相关的有效等位基因作用（即遗传基因影响）的加权和，可用于估计一个人出现某种生理/心理特征、罹患某种疾病的风险

## Article

- Shared genetic architecture between irritable bowel syndrome and psychiatric
  disorders reveals molecular pathways of the gut-brain axis
  <https://pubmed.ncbi.nlm.nih.gov/37528461/>

## Tool (Variants -> Phenotype)

### Direct discovery: GWAS -> SNP -> eQTL -> Genes

- Sherlock
  <https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3644637/>
- SMR
  <https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6346739/>

### Predication: GWAS -> SNP -> Genes -> Pathways and tissues

- NetWAS (GIANT websevier)
  <https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4828725/>
- Depict
  <https://www.nature.com/articles/ncomms6890>
  <https://data.broadinstitute.org/mpg/depict/>

(Although cell-type-specific expression quantitative trait loci (eQTLs) or coding (non-synonymous) variants in strong LD with associated variants can potentially link these variants to genes, overlap with eQTLs or coding variants may be coincidental.)

Direct functional follow-up of the many potentially causal variants and genes is typically difficult and expensive, so an attractive first step is to use computational approaches to prioritize genes in associated loci with respect to their likely biological relevance, and to identify pathways and tissues to define their likely biological context. 

### Polygenic overlap

For a pair of traits, polygenic overlap refers to the fraction of genetic variants causally associated with both traits over the total number of causal variants across the two traits considered, while genetic correlation quantifies the correlation coefficient of additive genetic effects for the two traits
(多基因重叠是指与两个性状因果相关的遗传变异占所考虑的两个性状的因果变异总数的比例)

- Mixer
  <https://github.com/precimed/mixer>

- MiXeR-Pred
  <https://github.com/precimed/mixer-pred>

- ConjFDR
  Improved Detection of Common Variants Associated with Schizophrenia by
  Leveraging Pleiotropy with Cardiovascular-Disease Risk Factors

# DNA 甲基化

<http://www.360doc.com/content/21/0118/13/72917688_957595208.shtml>
<https://mp.weixin.qq.com/s?__biz=MzAxMDkxODM1Ng==&mid=2247492739&idx=1&sn=c044bb55fe19d48f3b8299e3b41949d6&scene=21#wechat_redirect>

基因沉默相关

肿瘤中普遍存在DNA甲基化状态的改变，其特点是总体甲基化水平的降低与局部甲基化水平的升高。在肿瘤细胞中，癌基因处于低甲基化状态而被激活，抑癌基因处于高甲基化状态而被抑制。

## tool

[@BenchmarkingDnGong2022]

- Abismal (bisulfite sequencing in DNA methylation studies)
  <https://github.com/smithlabcode/abismal>
- BSMAP
- bismark (dep)

# RNA 甲基化

<https://mp.weixin.qq.com/s?__biz=MzU1MzMxMzcyMg==&mid=2247637352&idx=5&sn=f2a4da929cc6b7db289aad418fa1b86f&chksm=fbf82dbdcc8fa4ab8ea3dfed0f81c6296e00c5121001492120aaec624d0eb21f4970d77d2774&scene=27>

目前知道的最普遍的RNA甲基化修饰是甲基腺嘌呤N6甲基化（m6A）。mRNA的m6A修饰会影响mRNA本身的稳定性及其翻译成蛋白质的效率；最近的研究也表明RNA-m6A修饰还可逆向影响染色质重塑和组蛋白修饰等。

- Reading, writing and erasing mRNA methylation [@ReadingWritinZaccar2019]

# Metagenome

metagenomics assembler for PacBio HiFi reads
- <https://github.com/GaetanBenoitDev/metaMDBG?tab=readme-ov-file#building-from-source-using-conda>

<https://zhuanlan.zhihu.com/p/105602400>

# PlantDB

## Database

<https://plantaedb.com/>
<https://bidd.group/NPASS/index.php>
<https://plantaedb.com/taxa/phylum/gymnosperms/order/cupressales/family/taxaceae/genus/taxus/species/taxus-mairei>

# RNA binding

## Platform

[@TheBiochemicalMcgear2019]
The biochemical basis of microRNA targeting efficacy
<https://www.targetscan.org/vert_80/>
<https://www.targetscan.org/cgi-bin/targetscan/data_download.vert80.cgi>

starBase v2.0: decoding miRNA-ceRNA, miRNA-ncRNA and protein-RNA interaction networks from large-scale CLIP-Seq data
<https://rnasysu.com/encori/>

[@Enrigh2003]
miRanda is an algorithm for finding genomic targets for microRNAs. This algorithm has been written in C and is available as an open-source method under the GPL. MiRanda was developed at the Computational Biology Center of Memorial Sloan-Kettering Cancer Center.
`conda install -c bioconda miranda`
<https://bioconda.github.io/recipes/miranda/README.html>
<https://bioweb.pasteur.fr/packages/pack@miRanda@3.3a>
<http://www.microrna.org/>
<https://cbio.mskcc.org/miRNA2003/miranda.html>


# Metabolites

## database

- metabolites-GWAS (blood metabolites, SNP)
  [@AnAtlasOfGenShin2014]
  <https://pubmed.ncbi.nlm.nih.gov/24816252/>
  <http://metabolomics.helmholtz-muenchen.de/gwas/>
  [@MendelianRandoLiuX2022]
  Mendelian randomization analyses support causal relationships between blood metabolites and the gut microbiome
  <https://pubmed.ncbi.nlm.nih.gov/34980918/>

- metabolites-gut microbiome
  [@MendelianRandoLiuX2022]

# Database

## 16s-RNA Colorectal cancer

<https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE163366>
<https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE165255>

## GWAS

- open gwas
<https://www.bioconductor.org/packages/release/bioc/html/MungeSumstats.html>
<https://www.bioconductor.org/packages/release/bioc/vignettes/MungeSumstats/inst/doc/OpenGWAS.html>
<https://gwas.mrcieu.ac.uk/datasets/>

## Proteo-genomic convergence of human diseases

Mapping the proteo-genomic convergence of human diseases
<https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9904207/>

## GTEx (Genotype-Tissue Expression) Project

### Website

<https://gtexportal.org/home/>
<https://gtexportal.org/home/downloads/adult-gtex>

- WES / WGS
- RNA-seq, tissue

#### Extension data

- GPEdit: the genetic and pharmacogenomic landscape of A-to-I RNA editing in cancers
    - <https://hanlab.uth.edu/GPEdit/>

### Available

- QTL-seq
- RNA-seq
- Single-cell
- variants

### API 

<https://gtexportal.org/home/apiPage>
<https://gtexportal.org/api/v2/redoc#tag/GTEx-Portal-API-Info>

### example article

- RNA editing underlies genetic risk of common inflammatory diseases
  [@RnaEditingUndLiQi2022]
  (Here we show that ADAR-mediated adenosine-to-inosine RNA editing, a
  post-transcriptional event vital for suppressing cellular double-stranded RNA
  (dsRNA)-mediated innate immune interferon responses5-11, is an important
  potential mechanism underlying genetic variants associated with common
  inflammatory diseases)

<https://news.bioon.com/article/dac2e3647916.html>

# Some packages

## simulation

Learning Single-Cell Perturbation Responses using Neural Optimal Transport
<https://github.com/bunnech/cellot>

## Disease and mutation

VarGen: an R package for disease-associated variant discovery and annotation
<https://github.com/MCorentin/VarGen>

# Single cell tools

## article

- Cytotrace To predict the relative differentiation state of cells, we performed CytoTRACE (v0.1.0) analysis based on the expression data in monocyte sub-clustering results.
- AUCell analysis To determine whether activation of certain pathways differs between cells, the AUCell R package was utilized to calculate the degree of pathway activation for each cell using gene set enrichment analysis
- PALMO To discover how genes change with age, we performed PALMO (v0.1.2) analysis55 based on the expression data in immune cell clusters annotated as monocytes.

## Nature biotech

## Used

- Robust integration of multiple single-cell RNA sequencing datasets using a
  single reference space
  <https://github.com/bioinfoDZ/RISC>

## Novel

- De novo detection of somatic mutations in high-throughput single-cell profiling data sets
  [@DeNovoDetectiMuyas2023]
  <https://github.com/cortes-ciriano-lab/SComatic>

- PhyloVelo enhances transcriptomic velocity field mapping using monotonically.
  <https://github.com/kunwang34/PhyloVelo>
  <https://phylovelo.readthedocs.io/en/latest/index.html>
  (expressed genes Single-cell RNA
  sequencing (scRNA-seq) is a powerful approach for studying cellular
  differentiation, but accurately tracking cell fate transitions can be
  challenging, especially in disease conditions)

- A relay velocity model infers cell-dependent RNA velocity
  (a scalable deep neural network that locally infers velocity for each cell
  from its neighbors and then relays a series of local velocities to provide
  single-cell resolution inference of velocity kinetics)
  <https://github.com/GuangyuWangLab2021/cellDancer>
  <https://guangyuwanglab2021.github.io/cellDancer_website/>

- RNA velocity of single cells (Here we show that RNA velocity—the time
  derivative of the gene expression state—can be directly estimated by
  distinguishing between unspliced and spliced mRNAs in common single-cell RNA
  sequencing protocols)
  <https://github.com/velocyto-team/velocyto.R>


# MR (Mendelian randomization)

## concept

- Exposure factor (e.g., metabolites associated GWAS)
- results event (disease associated GWAS)
- OR <https://www.jianshu.com/p/edfea31b9093>
- RR <https://zhuanlan.zhihu.com/p/112866281>

## article

<https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7905253/>

## tools

R TwoSampleMR
<https://github.com/MRCIEU/TwoSampleMR>
<https://mrcieu.github.io/TwoSampleMR/index.html>

## workflow

<https://zhuanlan.zhihu.com/p/643280563>
- gwas
- 强工具变量 SNP
- 去除连锁不平衡
- 去除弱工具变量
- 去除混杂因素
- MR
- 异质性分析 (数据不同来源的影响)
- 多效性分析 (暴露因素以外的其它因素)
- 敏感性分析 leave-one-out 法 (指的是逐步剔除每个SNP，计算剩余SNP的meta效应，观察剔除每个SNP后结果是否发生变化)

## Method

MR方法选择三大原则：

（1）在没有异质性和多效性的情况下，优先使用IVW的估计结果；

（2）只有异质性，没有多效性时，优先使用Weighted Median方法的结果（也可以使用IVW的随机效应模型）；

（3）有多效性时，优先使用MR-Egger方法计算出的结果。

# RNA editing

## RNA editing database

- DARNED (limited infomation)
    - <https://darned.ucc.ie/>
    - <https://darned.ucc.ie/static/downloads/hg19.txt>
- RADAR <http://rnaedit.com/>
- GPEdit (cancer) <https://hanlab.uth.edu/GPEdit/>

## RNA editing method


- Predicting prime editing efficiency and product purity by deep learning.

about prime editing (PE)
- Enhanced prime editing systems by manipulating cellular determinants of editing outcomes

- Modular, programmable RNA sensing using ADAR editing in living cells
  <https://github.com/kristjaneerik/radar-rna-sensing>

- Precise integration of large DNA sequences in plant genomes using PrimeRoot editors

- Sequence-specific prediction of the efficiencies of adenine and cytosine base editors

## RNA editing

[@TheRoleOfRegAlbert2015]
**eQTL**: SNPs $\Rightarrow$ mRNA expression levels
- cis-eQTLs, local gene
- trans-eQTLs, distant gene

[@RnaSplicingIsLiYa2016]
**sQTL**: SNPs $\Rightarrow$ mRNA splicing
- RNA splicing is a primary link between genetic variation and disease

[@RnaEditingUndLiQi2022]
- RNA editing as an important mechanism underlying the genetic risk for
  numerous autoimmune and immune-related diseases
- ADAR1–dsRNA–MDA5 axis as an underlying mechanism in immune disease
- the extent to which common genetic differences in RNA editing may contribute
  to immune and inflammatory diseases remains to be identified
- eQTLs
- edQTLs
- sQTLs
- dsRNA
- Previous studies have identified millions of RNA editing sites in humans,
  more than 99% of which are located in **inverted repeat Alus** (IRAlus) that form
  dsRNA substrates

[@FastAndEfficiOngen2016]
- Fast and efficient **QTL mapper** for thousands of molecular phenotypes.
  <https://github.com/francois-a/fastqtl>
  <http://fastqtl.sourceforge.net/>

[@RnaEditingExBooth2023]
- ADAR

[@TargetedRnaEdXuWe2022]
Mode:
- **A-to-I editing** (mostly)
    - A-to-I editing events on tRNAs are carried out by ADATs (Adenosine deaminase
      that act on tRNA, (Torres et al., 2014)
    - Whereas ADARs (Adenosine deaminases that act on RNA) are solely responsible
      for A-to-I editing on mRNAs (Melcher et al., 1996) and other non-coding RNAs
- **Cytosine-to-Uracil (C-to-U) editing**
    - catalyzed by members of the APOBEC

Genetic Methods:
- **TRIBE**
- **HyperTRIBE**: TRIBE/HyperTRIBE is to date the most widely adopted genetic
  method to identify RBP targets.
- RNA-tagging
- APOBEC-based methods

Immunoprecipitation (identify RBP targets): 
- RIP
- Crosslinking immunoprecipitation (CLIP)
- CLIP coupled to high-throughput sequencing (HITS) is referred to as HITS-CLIP
- ...
- Immunoprecipitation requires dependable and specific anti-RBP antibodies

Detection of transcriptome-wide microRNA-target interactions in single cells with **agoTRIBE**


[@RnaModificatioCuiL2022]

[@ClusterGuideRReauts2022]
- method

## Concept

- QTL mapping
    - Fast and efficient QTL mapper for thousands of molecular phenotypes.
      <http://fastqtl.sourceforge.net/>
    - <https://zhuanlan.zhihu.com/p/617962170>

- RNA transcription
    - m6A modification
    - ...
- RNA splicing
    - m6A modification
    - ...

## RNA modification

[@RnaM6aMethylaSendin2023]

[@RnaModificatioHanD2023]
- **MeRIP-seq** (methylated RNA immunoprecipitation sequencing)
- The first disadvantage is that these methods cannot be used for
  quantification of the modification ratio.
- **MAZTER-seq** and **m6 A-REF-seq** (m6 A-sensitive
  RNA-endoribonuclease-facilitated sequencing), use the MazF RNase enzyme that
  selectively cleaves RNA at unmethylated ACA motifs but not their m6 A
  methylated counterparts.
- A second limitation faced when using antibody-based RNA modification
  profiling methods is poor sensitivity.
- A third disadvantage of antibody-based profiling methods is that these
  methods cannot reveal the location of a modified site with single-base
  resolution.

[@ScdartSeqReveTegows2022]
- **scDART-seq** (deamination adjacent to RNA modification target se- quencing)
  successfully monitors m6 A modification at the single-cell level

[@DartSeqAnAnMeyer2019]
- **DART-seq**: an antibody-free method for global m6A detection

[@M6aRnaModificHuLu2022]
- a method for quantitative, whole-transcriptome mapping of m6A at
  single-nucleotide resolution
- **m6A-SAC-Seq**

[@TheRoleOfM6aSunT2019] 

# Network method

Network-based approaches for modeling disease regulation and progression [@NetworkBasedAGalind2023]

- SigMod: <https://github.com/YuanlongLiu/SigMod>.
- NeDRex: <https://api.nedrex.net/> (drug selection)
- BiCoN: <https://pypi.org/project/bicon> <https://exbio.wzw.tum.de/bicon> (drug selection)

Gene regulatory network inference
- DeepDRIM: <https://github.com/jiaxchen2-c/DeepDRIM>

Differential co-expression networks

- DICER: <http://acgt.cs.tau.ac.il/dicer/> (JAVA tools)
- Differential co-expression-based detection of conditional relationships in transcriptional data: comparative analysis and application to breast cancer

# Prediction

- Machine learning for genetics-based classification and treatment response prediction in cancer of unknown primary

- Learning single-cell perturbation responses using neural optimal transport
  <https://www.nature.com/articles/s41592-023-01969-x>
  <https://www.research-collection.ethz.ch/handle/20.500.11850/612005>
  <https://github.com/bunnech/cellot>

- (GWAS) Leveraging polygenic enrichments of gene features to predict genes underlying complex traits and diseases 
  <https://github.com/FinucaneLab/pops>

# A design

## Item

- SCSA 结果读取转化
- 分析方法模块化（结果展现）


# Excellent article

[@SingleCellRnaPeng2019]
Single-cell RNA-seq highlights intra-tumoral heterogeneity and malignant
progression in pancreatic ductal adenocarcinoma
<https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6796938/>

- Unveiling the role of AGT in lipid metabolism and regulated cell death in colon cancer
<https://pmc.ncbi.nlm.nih.gov/articles/PMC11214316/>
